### PR TITLE
Removing strong references to to Context

### PR DESCRIPTION
**Pull request by sojanpr**
Strong reference to the context variable may create mem leak depending
on how activities are handled in the app implementation.

@aaustin @Sarkar

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/ShareLinkManager.java [hunk 1]
```diff
@@ -319,31 +322,31 @@ public boolean isEnabled(int position) {
...
-            this.setMinWidth(context_.getResources().getDisplayMetrics().widthPixels);
+            this.setMinWidth(context_.get().getResources().getDisplayMetrics().widthPixels);
         }
 
         public void setLabel(String appName, Drawable appIcon, boolean isEnabled) {
             this.setText("\t" + appName);
             this.setTag(appName);
             if (appIcon == null) {
-                this.setTextAppearance(context_, android.R.style.TextAppearance_Large);
+                this.setTextAppearance(context_.get(), android.R.style.TextAppearance_Large);
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/ShareLinkManager.java]: for these 3, should we check if not null before trying to use it?

Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/ShareLinkManager.java]: also, they aren't using the ShareLinkBuilder, so I'd say we shouldn't over correct this class. Thoughts?

Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/ShareLinkManager.java]: Agree. Will Add the null check before getting anything using weak reference

Comment (Sarkar) [inline Branch-SDK/src/io/branch/referral/ShareLinkManager.java]: i think ShareLinkBuilder had a memory leak before, so it's better to account for this especially if we're making Activity reachable by weak reference in Branch.java anyway

**Comment by Sarkar** [comment]
Nice tactic using WeakReference - I think this is a good way to not leak on Activity :+1:

File: Branch-SDK/src/io/branch/referral/ShareLinkManager.java (hunk 1 above)
Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/ShareLinkManager.java]: Figured out how exactly the leak happens with ShareLinkManager and added  fix for the issue. Removed previous corrections with weak reference for the share link manager

**Comment by sojanpr** [comment]
Moved to  PR https://github.com/BranchMetrics/Android-Deferred-Deep-Linking-SDK/pull/216

**Pull request closed without merging**